### PR TITLE
[minitunnel] track tunneled ports for listing and closing

### PR DESCRIPTION
**Pull request by activeshadow**
Track and display port forward tunnels created using miniccc, and allow for existing tunnels to be torn down.

### CODE REVIEW

**Comment by jacdavi** [comment]
Yep I can take a look next week. Trying to get version 2.8 released atm, so this will be the first change for 2.9 :)

**Comment by activeshadow** [comment]
Ah shoot... that means this one and the one I created earlier today missed the cutoff. 😞

File: cmd/minimega/cc_cli.go
Comment (jacdavi) [inline cmd/minimega/cc_cli.go:1]: Could you add a small blurb in the long help about `cc tunnel`?

File: cmd/minimega/cc_cli.go [hunk 1]
```diff
@@ -236,6 +238,85 @@ func cliCC(ns *Namespace, c *minicli.Command, resp *minicli.Response) error {
...
+
+	if c.BoolArgs["list"] {
+		// map VM name --> VM UUID
+		vms := make(map[string]string)
+
+		if v == Wildcard {
+			clients := ns.ccServer.GetClients()
+
+			for _, client := range clients {
+				vms[client.Hostname] = client.UUID
```
Comment (jacdavi) [inline cmd/minimega/cc_cli.go:270]: Small issue I noticed here. I had two VMs booted that had the same hostname, so they clobbered each other, and only 1 showed. It's also slightly inconsistent that `all` outputs hostnames, while you need the actual vm name or UUID when specifying a specific vm.

Maybe we do an extra lookup to convert from UUID -> name then use that as the key?

**Review by jacdavi — COMMENTED** [review]
One final thing. On closing a tunnel, I consistently get error outputs from the closed connection

```
2023/08/24 19:31:30 ERROR minitunnel.go:219: accept tcp [::]:9023: use of closed network connection
2023/08/24 19:31:30 ERROR minitunnel.go:303: read tcp [::1]:9023->[::1]:39264: use of closed network connection
```
(only got the 2nd line once)

Is it possible to suppress those errors when the user explicitly closed the tunnel?

---

Otherwise this is really cool! I hadn't used `cc tunnel` before, and I think if I had, I would've expected these features to already be there.

File: cmd/minimega/cc_cli.go
Comment (activeshadow) [reply cmd/minimega/cc_cli.go:1]: Please take a look at my latest commits that attempt to address this and your comment above.

File: cmd/minimega/cc_cli.go (hunk 1 above)
Comment (activeshadow) [reply cmd/minimega/cc_cli.go]: My latest commits should address this as well.

**Review by jacdavi — APPROVED** [review]

**Comment by jacdavi** [comment]
Looks good! Thanks for the changes!

**Pull request merged**